### ISSUE TITLE
Multiline Destructuring Assignment breaks syntax highlighting

### ISSUE THREAD

**Issue by vilic** [Microsoft/TypeScript-TmLanguage#116]
```ts
class Foo {
    bar() {
        let {

        } = yo;

        `${biu}`;
    }
}
```

![image](https://cloud.githubusercontent.com/assets/970430/13605492/bc33ea30-e582-11e5-8728-e2587e1ef1e4.png)

**Comment by billti** [comment]
@anubmat Is this caused by any of the recent fixes?

Does this repro when the template string is part of a larger expression (e.g. assignment, argument list, etc.)? If it only repros as a standalone statement expression it seems less critical, otherwise we should get this fixed ASAP.

**Comment by anubmat** [comment]
The template string is not causing the problem here. It is being caused by the multiline let expression
`let{`
`}`
which breaks coloring.
![error_116](https://cloud.githubusercontent.com/assets/17298501/13620505/2c97607e-e545-11e5-9819-8413fbe31ab1.PNG)
I don't think it is caused by any of the recent fixes, but I will check again.

**Comment by TheTedAdams** [comment]
I opened this same issue over at atom-typescript not realizing it should be here: https://github.com/TypeStrong/atom-typescript/issues/874

Multiline destructuring the opening bracket seems to not get counted as a bracket by syntax highlighter, so the closing brack makes the syntax highlighter think you are outside of the surrounding method (it seems to me)

**Comment by xealot** [comment]
I believe this also happens when importing multiline ES6 imports...

```javascript
import {A, B, C, 
    D} from "./types";
```

**Comment by sheetalkamat** [comment]
Fixed by #257
